lib/builtin.t

```diff
@@ -176,4 +176,79 @@ package FetchStoreCounter {
     cmp_ok($val, $_,  !1, "false is equivalent to  !1 by $_") for qw( eq == );
 }
 
+# Vanilla trim tests
+{
+    use builtin qw( trim );
+
+    is(trim("    Hello world!   ")      , "Hello world!"  , 'Trim spaces');
+    is(trim("\tHello world!\t")         , "Hello world!"  , 'Trim tabs');
+    is(trim("\n\n\nHello\nworld!\n")    , "Hello\nworld!" , 'Trim \n');
+    is(trim("\t\n\n\nHello world!\n \t"), "Hello world!"  , 'Trim all three');
+    is(trim("Perl")                     , "Perl"          , 'Trim nothing');
+    is(trim('')                         , ""              , 'Trim empty string');
+}
+
+TODO: {
+    my $warn = '';
+    local $SIG{__WARN__} = sub { $warn .= join "", @_; };
+
+    is(builtin::trim(undef), ""                                        , 'Trim undef');
+    like($warn    , qr/^Use of uninitialized value in subroutine entry at/, 'Trim undef triggers warning');
+    local $main::TODO = "Currently uses generic value for the name of non-opcode builtins";
+    like($warn    , qr/^Use of uninitialized value in trim at/,
+         'Trim undef triggers warning using actual name of builtin');
+}
+
+# Fancier trim tests against a regexp and unicode
```

Comment:
I don't quite get the benefit of doing this set of tests. We've already tested `trim()` above, so why test it again as compared some function that only exists in this `.t` file anyway?

Reply:
I just cargo culted this code.  I see no point to it either.